### PR TITLE
Add support for creating feeds in genesis

### LINKED ISSUE #78: Extend feed pallet genesis config to create Feed
The current feed pallet genesis config only allows setting the admin and feed creators. 
Creating Feeds at genesis should be supported as well.

### PR DESCRIPTION
**Pull request by mattsse**
# Changes

- The GenesisConfig of the feed pallet allows for creating feeds during genesis

# Tests

`cargo t`

# Issues

- Closes #78

### CODE REVIEW

File: pallet-chainlink-feed/src/feed.rs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+use frame_support::{
+	sp_runtime::traits::{One, Zero},
+	Parameter,
+};
+use sp_std::prelude::*;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+/// Builder with all the parameters to call the `create_feed`
```
Comment (ebarakos) [inline pallet-chainlink-feed/src/feed.rs:13]: Any hints on how to test this @mattsse?

Comment (mattsse) [reply pallet-chainlink-feed/src/feed.rs:13]: @ebarakos This allows creating feeds that should be included in the genesis block with the feed pallet's genesisconfig:

https://github.com/smartcontractkit/chainlink-polkadot/pull/79/files#diff-c5eb2db45eadd23bc84e52e146f5af5c01907ae3ccd2b563f928ee67b565490aR169-R173

This builder helps creating those like in this test

https://github.com/smartcontractkit/chainlink-polkadot/pull/79/files#diff-34ba86f45aaa350811973171ddae78d326f79cb5b184943e213971dd291f20b0R1252-R1272

This allows creating feeds directly in the chainspec:
https://github.com/smartcontractkit/chainlink-polkadot/blob/master/substrate-node-example/node/src/chain_spec.rs#L163-L165

**Comment by mattsse** [comment]
Update:
- added chain spec json files that make it easier to configure custom setups
- more info in the readme

With these json specs it will be easy to test several different setups:
* feed creators and admin with endowment
* feed creators, admin with endowment and feed at genesis
* feed creators and admin without endowment

**Review by ebarakos — CHANGES_REQUESTED** [review]
Nice @mattsse, much easier to configure + to have preconfigured setups.

Is there any chance we can remove the common bloat from json specs, so they are cleaner and more related to the actual needed configuration? Especially the upper part of specs and this big "code" hex.

**Comment by mattsse** [comment]
@ebarakos the large hex is the wasm runtime code, but the feed pallet config is determined by `chainlinkFeed` json value.
So what we could do is write a helper script that takes a `chainlinkFeed` object (from a separate json) and plugs in into the chainspec?

Alternatively what would probably be the more convenient solution is, that we try to decode the given chain spec file as `chanlinkFeed` json and use this for the genesis config, for example: `./node-template --chain=chainlinkfeed.json --alice`

**Comment by ebarakos** [comment]
> @ebarakos the large hex is the wasm runtime code, but the feed pallet config is determined by `chainlinkFeed` json value.
> So what we could do is write a helper script that takes a `chainlinkFeed` object (from a separate json) and plugs in into the chainspec?
> 
> Alternatively what would probably be the more convenient solution is, that we try to decode the given chain spec file as `chanlinkFeed` json and use this for the genesis config, for example: `./node-template --chain=chainlinkfeed.json --alice`

I think both solutions should be good, just to be able to pass the different `chainlinkFeed` object/json/file as argument in command line.

**Comment by mattsse** [comment]
> I think both solutions should be good, just to be able to pass the different `chainlinkFeed` object/json/file as argument in command line.

as described in the specs/README.md the `--chain` option now accepts also a json string of a the feed pallet config or a path to a file like `specs/feedconfig.json`

**Review by ebarakos — APPROVED** [review]
Tested both in chain spec + as a command parameter in the demo's docker image and it works as expected.

**Pull request merged**